utilities/src/main/java/io/syndesis/qe/endpoints/publicendpoint/EnvironmentsPublicEndpoint.java

```diff
@@ -0,0 +1,65 @@
+package io.syndesis.qe.endpoints.publicendpoint;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.syndesis.common.util.Json;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
+@Component
+@Lazy
+public class EnvironmentsPublicEndpoint extends PublicEndpoint {
+
+    public EnvironmentsPublicEndpoint() {
+        super("/environments");
+    }
+
+    /**
+     * Get all tags in the syndesis
+     * endpoint -> GET ​/public​/environments
+     * original method -> {@link io.syndesis.server.endpoint.v1.handler.external.PublicApiHandler#getReleaseEnvironments()}
+     */
+    public List<String> getAllEnvironments() throws IOException {
```

Comment:
rather than throwing IOException it would be better to catch it here and fail the test (and then all methods that use `getAllEnvironments` do not need to handle it)

Reply:
In the new commit with the REST tests, I use `response.readEntity(new GenericType<List<String>>() {})` without IOException. I update it here because I have not already done REST tests